packages/camera/camera_windows/windows/camera.cpp

```diff
@@ -0,0 +1,271 @@
+    flutter::BinaryMessenger *messenger, int64_t camera_id) {
+  auto channel_name =
+      std::string(kCameraMethodChannelBaseName) + std::to_string(camera_id);
+  return std::make_unique<flutter::MethodChannel<>>(
+      messenger, channel_name, &flutter::StandardMethodCodec::GetInstance());
+}
+
+}  // namespace
+
+CameraImpl::CameraImpl(const std::string &device_id)
+    : device_id_(device_id),
+      capture_controller_(nullptr),
+      messenger_(nullptr),
+      camera_id_(-1),
+      Camera(device_id) {}
+CameraImpl::~CameraImpl() {
+  capture_controller_ = nullptr;
+
+  SendErrorForPendingResults("Plugin disposed",
+                             "Plugin disposed before request was handled");
+}
+
+void CameraImpl::InitCamera(flutter::TextureRegistrar *texture_registrar,
+                            flutter::BinaryMessenger *messenger,
+                            bool enable_audio,
+                            ResolutionPreset resolution_preset) {
+  auto capture_controller_factory =
+      std::make_unique<CaptureControllerFactoryImpl>();
+  InitCamera(std::move(capture_controller_factory), texture_registrar,
+             messenger, enable_audio, resolution_preset);
+}
+
+void CameraImpl::InitCamera(
+    std::unique_ptr<CaptureControllerFactory> capture_controller_factory,
+    flutter::TextureRegistrar *texture_registrar,
+    flutter::BinaryMessenger *messenger, bool enable_audio,
+    ResolutionPreset resolution_preset) {
+  assert(!device_id_.empty());
+  messenger_ = messenger;
+  capture_controller_ =
+      capture_controller_factory->CreateCaptureController(this);
+  capture_controller_->InitCaptureDevice(texture_registrar, device_id_,
+                                         enable_audio, resolution_preset);
+}
+
+// Adds pending result to the pending_results map.
+// If result already exists, call result error handler
+bool CameraImpl::AddPendingResult(
+    PendingResultType type, std::unique_ptr<flutter::MethodResult<>> result) {
+  assert(result);
+
+  auto it = pending_results_.find(type);
+  if (it != pending_results_.end()) {
+    result->Error("Duplicate request", "Method handler already called");
+    return false;
+  }
+
+  pending_results_.insert(std::make_pair(type, std::move(result)));
+  return true;
+}
+
+std::unique_ptr<flutter::MethodResult<>> CameraImpl::GetPendingResultByType(
+    PendingResultType type) {
+  if (pending_results_.empty()) {
+    return nullptr;
+  }
+  auto it = pending_results_.find(type);
+  if (it == pending_results_.end()) {
+    return nullptr;
+  }
+  auto result = std::move(it->second);
+  pending_results_.erase(it);
+  return result;
+}
+
+bool CameraImpl::HasPendingResultByType(PendingResultType type) {
+  if (pending_results_.empty()) {
+    return nullptr;
+  }
+  auto it = pending_results_.find(type);
+  if (it == pending_results_.end()) {
+    return false;
+  }
+  return it->second != nullptr;
+}
+
+void CameraImpl::SendErrorForPendingResults(const std::string &error_id,
```

Comment:
Seems like `dart run ./script/tool/bin/flutter_plugin_tools.dart format` does not fix these and my IDE currently does not use right style either.
I still need to fix this by configuring right formatter/linter to handle these proper way.

Reply:
I believe clang-format defaults to "prevailing style" for this, and we don't have an explicit setting. (It used to be impossible to have different settings for C++ and ObjC due to `.h` files, but I think that may be fixed now, in which case we could explicitly set it to type-align for C++ and name-align for ObjC.)

Reply:
Ok, I found out the reason why my clang-format was aligning pointers as name-align.
This is because clang-setting DerivePointerAlignment is set to true, and it just uses most common aligment of the file.
I'll fix all files with temporary .clang-format settings as:
```
BasedOnStyle: Google

Language: Cpp
# Force pointers to the type for C++.
DerivePointerAlignment: false
PointerAlignment: Left
```